x-pack/plugins/security_solution/public/common/lib/kuery/index.ts

```diff
@@ -161,9 +135,9 @@ const buildQueryMatch = (
         : checkIfFieldTypeIsDate(dataProvider.queryMatch.field, browserFields)
         ? convertDateFieldToQuery(dataProvider.queryMatch.field, dataProvider.queryMatch.value)
         : `${dataProvider.queryMatch.field} : ${
-            isNumber(dataProvider.queryMatch.value)
+            Array.isArray(dataProvider.queryMatch.value)
               ? dataProvider.queryMatch.value
-              : escapeQueryValue(dataProvider.queryMatch.value)
+              : prepareKQLParam(dataProvider.queryMatch.value)
```

Comment:
This is a functional change @maximpn -- is it a fix done on purpose? If yes, please add a unit test covering this, otherwise I'd suggest to revert it.

Reply:
The point is my change changes nothing. Confusion comes from `escapeQueryValue()` (which I removed) defined above. It handles only strings and returns any other values (a number, a boolean and an array of strings, numbers or booleans) as is which get stringified as used inside a string interpolation.